xdsl/irdl.py

```diff
@@ -1001,6 +1001,10 @@ def wrong_field_exception(field_name: str) -> PyRDLOpDefinitionError:
             # allowed in IRDL definitions.
             if parent_cls == Generic:
                 continue
+            if parent_cls is object:
```

Comment:
Why do we handle this here?
Should we instead handle this in `if field_name in _OPERATION_DICT_KEYS`?
Do we actually gain speedup from that?
If we don't, I would prefer keeping it simple to make it less bug-prone.

Reply:
as far as I know, `object` is the only class without `__annotations__`, annoyingly, so getattr fails. Seems like checking for this is better than hasattr annotations

Reply:
went away :)